conda/base/context.py

```diff
@@ -1154,6 +1162,17 @@ def cpu_flags(self):
         info = _get_cpu_info()
         return info["flags"]
 
+    @property
+    def plugins(self):
+        """Holds plugin related configuration settings"""
+        if self._plugins is None:
+            plugin_config_type = create_plugin_config_type(
+                self.plugin_manager.get_settings()
+            )
+            self._plugins = plugin_config_type(self.raw_data)
+
+        return self._plugins
+
```

Comment:
Dynamically creating types are a red flag in the year 2024, let's not do that. If you want to lazily load the plugins, just use `functools.cached_property`.

```suggestion
	from functools import cached_property

    @cached_property
    def plugins(self):
        """Holds plugin related configuration settings"""
        PluginType = create_plugin_config_type(self.plugin_manager.get_settings())
        return PluginType(self.raw_data)
```